### ISSUE TITLE
Support remaining sklearn make_* simulation functions

### ISSUE THREAD

**Issue by gbrener** [ContinuumIO/xarray_filters#5]
PR #2 introduces important sklearn-compatible functionality for xarray_filters, however according to the example notebook, some functions remain unsupported:
<img width="922" alt="screen shot 2017-09-06 at 4 31 15 pm" src="https://user-images.githubusercontent.com/2840348/30135581-fb59b65c-9320-11e7-8240-ba4594002764.png">
...
<img width="936" alt="screen shot 2017-09-06 at 4 31 30 pm" src="https://user-images.githubusercontent.com/2840348/30135589-00d252ba-9321-11e7-95d7-9620cc029177.png">
At some point we may want to make the API more generalizable so that these functions are supported.

**Comment by gpfreitas** [comment]
I'll wait for #PR 3 to be merged to address this.

**Comment by PeterDSteinberg** [comment]
Thanks for making this reminder issue.  Let's sequence this work after the larger Elm / Earthio refactoring of [Elm issue #149](https://github.com/ContinuumIO/elm/issues/149)'s next steps.  I'll add deferred label.  It's great that we'll get to full support for `sklearn.datasets` with `xarray` and `pandas` wrappers, but lower priority than the planning we need to do around ML refactor and unifying dask-* ML experiments.

cc @gbrener @gpfreitas

**Comment by gpfreitas** [comment]
Out of all the 4 functions above, none returns a pair of features/labels.

`make_spd_matrix` returns a simple sparse matrix (that is useful). That could be supported easily in our framework just allowing `y=None`, or that the sampling functions (`make_*`) return X1, X2, X3, ... where the first dimension coincides.

All others return 3 separate outputs:

- `make_biclusters` and `make_checkerboard` returns a triple of matrices for biclustering exercises (a matrix, and an indicator for the rows, as well as one for the columns)
- `make_sparse_coded_signal` also returns a triple, but it's meaning is completely different from every other sampling function in `sklearn.datasets`.

It would be nice to know if those three problematic functions are going to be of any use for elm. If not, I say we skip it for the foreseeable future.

cc @PeterDSteinberg, @gbrener
